### ISSUE TITLE
Fly 5.4.0 DNS Resolution issue

### ISSUE THREAD

**Issue by eliotstocker** [concourse/concourse#4149]
## Bug Report

After updating to concourse 5.4.0 and fly 5.4.0 fly can no longer resolve local cluster DNS records in Kubernetes. (either using the concourse pipeline resource or fly directly within the cluster)

## Steps to Reproduce

attempt to login against a local cluster DNS URL using Fly v5.4.0

## Expected Results

when running the command:
`fly -t ci login -c http://concourse-web.default.svc.cluster.local:8080`

expect (as of v5.3.0):
```
logging in to team 'main'

navigate to the following URL in your browser:

  http://concourse-web.default.svc.cluster.local:8080/login?fly_port=46637

or enter token manually:
```

## Actual Results

when running the command (v5.4.0):
`fly -t ci login -c http://concourse-web.default.svc.cluster.local:8080`

got (as of v5.4.0):
```
logging in to team 'main'

could not reach the Concourse server called ci:

    Get http://concourse-web.default.svc.cluster.local:8080/api/v1/info: dial tcp: lookup concourse-web.default.svc.cluster.local: no such host

is the targeted Concourse running? better go catch it lol
```

## Additional Context

* This occurs when running from inside a resource or job container and trying to connect to a local concourse web service.
* We can ping and curl the local DNS record from the same containers, so the issue appears to be specific to fly
* Fly v5.3.0 does not appear to present the same issue, it looks like the DNS resolution has changed in Fly 5.4.0

## Version Info

* Concourse version: 5.4.0
* Deployment type (BOSH/Docker/binary): Kubernetes/Helm
* Infrastructure/IaaS: GKE
* Browser (if applicable): NA
* Did this used to work? Yes

**Comment by kcmannem** [comment]
we're aware of this, working to fix it. You can follow the updates here https://github.com/concourse/concourse/issues/4141

**Comment by mgruener** [comment]
We have the same problem and I can confirm, doing `export GODEBUG=netdns=go` before calling fly does solve the problem. Or at least the negative effects.

**Comment by pivotal-jamie-klassen** [comment]
Fixed by concourse/ci#59
